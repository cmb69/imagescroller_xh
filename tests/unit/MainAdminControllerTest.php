<?php

/*
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Imagescroller_XH.
 *
 * Imagescroller_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Imagescroller_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Imagescroller_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Imagescroller;

use ApprovalTests\Approvals;
use Imagescroller\Infra\Repository;
use Imagescroller\Infra\View;
use Imagescroller\Value\Image;
use PHPUnit\Framework\TestCase;

class MainAdminControllerTest extends TestCase
{
    public function testRendersSelectboxAndGallery(): void
    {
        $_GET = ["imagescroller_gallery" => "gallery2"];
        $repository = $this->createMock(Repository::class);
        $repository->method("find")->willReturn($this->images());
        $repository->method("findAll")->willReturn(["gallery1", "gallery2", "gallery3"]);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["imagescroller"]);
        $sut = new MainAdminController($repository, $view);
        ob_start();
        $sut->defaultAction();
        $response = ob_get_clean();
        Approvals::verifyHtml($response);
    }

    private function images(): array
    {
        return [
            new Image("image1"),
            new Image("image2"),
            new Image("image3"),
        ];
    }
}
