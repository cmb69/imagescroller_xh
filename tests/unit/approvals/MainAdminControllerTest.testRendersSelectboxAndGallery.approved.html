<select onchange="window.document.location.href = '?&imagescroller&amp;admin=plugin_main&amp;imagescroller_gallery='+this.value">
  <option value="gallery1" >gallery1</option>
  <option value="gallery2" selected>gallery2</option>
  <option value="gallery3" >gallery3</option>
</select>
<form action="?imagescroller&amp;admin=plugin_main" method="post">
  <table>
    <tbody>
      <tr>
        <td>
          <img src="image1" width="200" height="" alt="">
          <input type="hidden" name="imagescroller_image[]" value="image1">
        </td>
        <td>
          <input type="text" name="imagescroller_title[]">
          <input type="text" name="imagescroller_desc[]">
          <input type="text" name="imagescroller_link[]">
        </td>
      </tr>
      <tr>
        <td>
          <img src="image2" width="200" height="" alt="">
          <input type="hidden" name="imagescroller_image[]" value="image2">
        </td>
        <td>
          <input type="text" name="imagescroller_title[]">
          <input type="text" name="imagescroller_desc[]">
          <input type="text" name="imagescroller_link[]">
        </td>
      </tr>
      <tr>
        <td>
          <img src="image3" width="200" height="" alt="">
          <input type="hidden" name="imagescroller_image[]" value="image3">
        </td>
        <td>
          <input type="text" name="imagescroller_title[]">
          <input type="text" name="imagescroller_desc[]">
          <input type="text" name="imagescroller_link[]">
        </td>
      </tr>
    </tbody>
  </table>
  <input type="hidden" name="action" value="save">
  <input type="submit" class="submit">
</form>
